pipeline/cactus_workflow.py

```diff
@@ -440,50 +510,50 @@ def inverseJukesCantor(d):
     assert d >= 0.0
     return 0.75 * (1 - math.exp(-d * 4.0/3.0))
     
-class CactusCafPhase(CactusPhasesTarget):      
-    def run(self):
+class CactusCafPhase(CactusPhasesJob):      
+    def run(self, fileStore):
         if (not self.cactusWorkflowArguments.configWrapper.getDoTrimStrategy()) or (self.cactusWorkflowArguments.outgroupEventNames == None):
             setupFilteringByIdentity(self.cactusWorkflowArguments)
         #Setup any constraints
         if self.getPhaseIndex() == 0 and self.cactusWorkflowArguments.constraintsFile != None: #Setup the constraints arg
-            newConstraintsFile = os.path.join(self.getGlobalTempDir(), "constraints.cig")
+            newConstraintsFile = fileStore.getLocalTempFile()
             runCactusConvertAlignmentToCactus(self.cactusWorkflowArguments.cactusDiskDatabaseString,
                                               self.cactusWorkflowArguments.constraintsFile, newConstraintsFile)
-            self.phaseNode.attrib["constraints"] = newConstraintsFile  
-        if self.getOptionalPhaseAttrib("alignments", default="") != "":
+            self.phaseNode.attrib["constraintsID"] = fileStore.writeGlobalFile(newConstraintsFile, cleanup=False)
+        if "alignmentsID" in self.phaseNode.attrib:
             # An alignment file has been provided (likely from the
             # ingroup vs. outgroup blast stage), so just run caf using
             # that file
             assert self.getPhaseNumber() == 1
-            convertedAlignmentsFile = getTempFile(rootDir=self.getGlobalTempDir())
+            convertedAlignmentsFile = fileStore.getLocalTempFile()
             # Convert the cigar file to use 64-bit cactus Names instead of the headers.
-            runConvertAlignmentsToInternalNames(self.cactusWorkflowArguments.cactusDiskDatabaseString, self.phaseNode.attrib["alignments"], convertedAlignmentsFile, self.topFlowerName)
-            self.logToMaster("Converted headers of cigar file %s to internal names, new file %s" % (self.phaseNode.attrib["alignments"], convertedAlignmentsFile))
-            self.phaseNode.attrib["alignments"] = convertedAlignmentsFile
+            runConvertAlignmentsToInternalNames(self.cactusWorkflowArguments.cactusDiskDatabaseString, self.cactusWorkflowArguments.alignmentsID, convertedAlignmentsFile, self.topFlowerName)
```

Comment:
You might have to download the alignmentsID file here first, I think. Sorry if this is already done somewhere else I didn't notice.

It may sometimes work without this, since this branch isn't always taken. After all is said and done we should probably prune the blast stage paths, since there are a few completely separate ways sequences get blasted right now, and they could all be handled by a single ingroup/outgroup blast path.